wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+    parser.add_argument('--root_func', default=False, action='store_true')
+
+    args = parser.parse_args()
+    
+    if not args.plot:
+        # Disables the canvas from ever popping up
+        gROOT.SetBatch()
+    
+    canvas = []
+    canvas_count = 1
+    if os.path.exists('temp_data.csv'):
+        os.remove('temp_data.csv')
+
+    for filename in args.filenames:
+        f = ROOT.TFile(filename,"UPDATE")
+        keycount=0
+        
+        # raw is unfiltered
+        raw_histograms = {}
+        filtered_histograms = {}
+        for key in f.GetListOfKeys():
+            #print("***KEYPRINT")
+            #print(key)
+            #print(key.GetTitle())
+            if (keycount>0):
+                if(key.GetTitle()==keyold.GetTitle()):
+                   # print("EQUAL")
+                    continue
+            keycount+=1
+            keyold=key
+            #if keycount>2:
+            #    break
+
+            #if keycount>args.ObjKeep:# Only gets channel 1
+            #    break
+                # Argument to delete all previous fits. Not working, related to TF1 class?
+                # Each time the code executes 2 new keys are created in the file.
+                #h.Reset("ICESM")
+
+            h = key.ReadObj()
+
+            if h.ClassName() != 'TH1F' and h.ClassName() != 'TH1D':
+                continue
+
+            if h.GetName().startswith('f_'):
+                filtered_histograms[h.GetName()] = h
+                keycount -= 1
+            else:
+                raw_histograms[h.GetName()] = (h, keycount)
+        
+        for histogram in raw_histograms:
+            h = raw_histograms[histogram][0]
+            keycount = raw_histograms[histogram][1]
+            
+
+            print(h.GetName())
+            if args.sodium:
+                # Peak searching using the TSpectrum class. See ROOT
+                # documentation for details.
+                # https://root.cern.ch/root/htmldoc/guides/spectrum/Spectrum.html#processing-and-visualization-functions
+
+                # *** depending on the overvoltage the peak dimension with respect to the noise makes changes needed on the second value. A more efficient way would be to create a bisection and stop it when it detects only the 2 biggest peaks.
+                #The number of found peaks and their positions are written into the members fNpeaks and fPositionX
+                #Search 	( 	const TH1 *  	hin,
+	        #Double_t  	sigma = 2,
+		#Option_t *  	option = "",
+		#Double_t  	threshold = 0.05 #Peaks with  amplitude < th*Amax are discarded. If too small -> detects peaks after the 511 peak. The code right now detects the last peak (that doesn't have a huge width) as the 511 peak. Other option is to iterate through the y positions and keep always the latest peak with an increase over the previous ones? first derivative with respect to max/min closest to zero 	
```

Comment:
Formatting on the comments here? One nice thing you can do in vim is block highlight a section of code and then press "gq" which should auto format it to 80 characters per line.

Reply:
That is a great vim tip, thanks!